pkg/probes/curl_json/machine_images.go

```diff
@@ -0,0 +1,81 @@
+package curl_json
+
+import "github.com/openshift/osd-network-verifier/pkg/helpers"
+
+// cloudMachineImageMap is a lookup table mapping VM image IDs to their
+// respective cloud platforms, CPU architectures, and cloud regions. To
+// access, reference cloudMachineImageMap[$CLOUD_PLATFORM][$CPU_ARCH][$REGION];
+// e.g., cloudMachineImageMap[helpers.PlatformAWS][helpers.ArchX86]["us-east-1"]
+// Note that GCP images are global/not region-scoped, so the region key will
+// always be "*"
+var cloudMachineImageMap = map[string]map[string]map[string]string{
+	helpers.PlatformAWS: {
```

Comment:
I think this is fine for now, but I worry about how we'll handle Legacy vs. RHEL9 base images in the future. Would the idea be an entirely separate set of nested maps?

Reply:
Yup! In a totally separate file/directory. You can see how I did it for the legacy probe [here](https://github.com/openshift/osd-network-verifier/pull/241/files#diff-1f92ce6064310e73a3611adb32e0f3779e3dbd7abe1c7750c05486910b795ae2).